helm/charts/infra/templates/server/secret.yaml

```diff
@@ -0,0 +1,29 @@
+
+{{- if include "server.enabled" . | eq "true" }}
+{{- if (not .Values.server.config.tls) }}
+
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "server.fullname" . }}-ca
+  labels:
+{{- include "server.labels" . | nindent 4 }}
+data:
+
+{{- $secret := lookup "v1" "Secret" .Release.Namespace (printf "%s-ca" (include "server.fullname" .)) -}}
+{{- if $secret.data }}
+  ca.crt: |
+{{- get $secret.data "ca.crt" | nindent 4 }}
+  ca.key: |
```

Comment:
Can I specify the CA without specifying the CA's key? Or is this CA cert used to generate other certificates?

Reply:
In the default setup provided by these helm charts it generates both, so both will always be specified. A user can specify only the CA if they create their own cert and key, but in that case the CA only gets used to verify client certs for mTLS.